src/NHibernate/Transaction/AdoNetTransactionFactory.cs

```diff
@@ -136,7 +136,7 @@ public virtual void ExecuteWorkInIsolation(ISessionImplementor session, IIsolate
 					isolaterLog.Warn(ignore, "Unable to dispose transaction");
 				}
 
-				if (session.Factory.Dialect is SQLiteDialect == false)
+				if (connection != null && session.Factory.Dialect is SQLiteDialect == false)
```

Comment:
That's some messy method here.... Additional null check doesn't make it better. May be instead move connection retrieving logic  outside of `try` block?

Reply:
Well maybe not.. That would change exception..

Reply:
At least this will no more swallow a connection opening failure. But yes, it is indeed better to put the connection opening out of the try. It has no sense being into it.

Reply:
Well... Changing from NRE unrelated to the trouble to the bare exception is still better.
But yes, it loses the `HibernateException` encapsulation.

Switching back.

Reply:
I've just realized that this exception never was wrapped (my main concern was that we changed existing behavior - but we didn't) and I really don't see any benefits in such wrappings (seems like some legacy practice). So maybe better again switch back? )